### DISCUSSION TITLE
Load javascript saved via editor.getJs()

### DISCUSSION THREAD

**Discussion by Traver72** [GrapesJS/grapesjs#3641] [Q&A] [answered]
I store javascript via editor.getJs() in a database.
When loading the page, I fetch the javascript and try to insert it into the page with (allowScripts: 1 is enabled):

```
// page_content_js is the stored editor.getJs() 

this.editor.addComponents({
  type: 'script',
  content: page_content_js
})
```

But it doesn't work as expected. Only the last element added can be used as expected.
Is there a better way to load the javascript into the editor? Thanks a lot in advance.

**Comment by artf** [comment, marked as answer]
Hi @Traver72 as explained here https://grapesjs.com/docs/modules/Storage.html#store-and-load-templates you have to store/load the JSON. What you get from `getJs()`/`getHtml()`/`getCss()` is only to use for "final pages" as they don't have all the information necessary for the editor.

**Reply by Traver72** [reply]
Hello @artf,
thank you very much for the answer. I will restructure my code as explained on the page.
